### PR TITLE
Add legacy moniker migration logic

### PR DESCRIPTION
**Pull request by danieljurek**
Associated PRs depending on this change: 

* https://github.com/Azure/azure-sdk-for-python/pull/31955
* https://github.com/Azure/azure-sdk-for-js/pull/27032
* https://github.com/Azure/azure-sdk-for-java/pull/36649
* https://github.com/Azure/azure-sdk-for-net/pull/36194

This adds logic which transfers metadata JSON files and package-level overview pages from the `preview` and `latest` monikers to the `legacy` moniker. It also expands ToC and Service-level overview automation to the legacy moniker.

### CODE REVIEW

**Review by weshaggard — APPROVED** [review]

File: eng/common/scripts/Update-DocsMsPackageMonikers.ps1 [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    if (Test-Path $readmePath) {
+        Remove-Item $readmePath -Force
+    }
+}
+
+$metadataLookup = @{ 
+    'latest'  = getPackageMetadata 'latest'
+    'preview' = getPackageMetadata 'preview'
+}
+$deprecatedPackages = (Get-CSVMetadata).Where({ $_.Support -eq 'deprecated' })
```
Comment (danieljurek) [inline eng/common/scripts/Update-DocsMsPackageMonikers.ps1:84]: Also add not hidden.

Comment (danieljurek) [reply eng/common/scripts/Update-DocsMsPackageMonikers.ps1:84]: After further discussion, we're keeping the logic as written here.

**Comment by danieljurek** [comment]
Check statuses are stuck.

**Pull request merged**